glue_vispy_viewers/common/tests/test_vispy_toolbar.py

```diff
@@ -33,16 +33,16 @@ def callback(self, mode):
         self._called_back = True
 
 
-def test_toolbar():
+def test_toolbar(tmpdir):
     session = simple_session()
     with warnings.catch_warnings(record=True) as w:
         viewer = ExampleViewer(session)
         toolbar = viewer.toolbar
 
-        # TODO: how to add a test for dialogue option?
         # test save tool
-        # toolbar.actions['Save'].trigger()
-        # assert toolbar.active_tool.tool_id == 'Save'
+        with patch('qtpy.compat.getsavefilename') as fd:
+            fd.return_value = tmpdir.join('test.png').strpath, 'jnk'
+            toolbar.actions['vispy:save'].trigger()
```

Comment:
@PennyQ - just for info, this is how you can test things that use e.g. the save dialog. This replaces the normal save dialog function with a 'fake' one that just returns a filename.